### PR TITLE
add cursor implementation

### PR DESCRIPTION
**Pull request by DavidBM**

### CODE REVIEW

File: src/cursor.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    /// Loads all pages from the cursor and consumes it in the process.
+    pub async fn load_all(mut self) -> QLDBResult<DocumentCollection> {
+        let mut result = DocumentCollection::new(vec![]);
+
+        while let Some(values) = self.load_more().await? {
+            result.extend(values.into_iter());
+
+            if self.next_page.is_none() {
+                break;
+            }
```
Comment (aegis22) [inline src/cursor.rs:106]: Isn't this break covered by the condition of being Some(values) in the loading?

File: tests/cursor.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+#[async_std::test]
+async fn cursor_800_documents() {
+    let client = QLDBClient::default("rust-crate-test").await.unwrap();
+
+    let test_table = ensure_test_table(&client).await;
+
+    let documents_model = format!("cursor_800_documents{}", rand_string());
+
+    for _ in 0..20 {
```
Comment (aegis22) [inline tests/cursor.rs]: I think is better to use a well named variable than a magical number like 20.

File: tests/cursor.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        let table = test_table.clone();
+        let model = documents_model.clone();
+
+        client.transaction_within(|tx| async move {
+
+            let mut query = tx.query(&format!("INSERT INTO {} << ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? >>", &table));
+
+            for _ in 0..40 {
```
Comment (aegis22) [inline tests/cursor.rs]: Same as above with 40.

File: tests/transaction.rs [hunk 4]
```diff
@@ -81,11 +75,8 @@ async fn qldb_transaction_rollback() -> Result<()> {
 
                 let count = result?;
 
-                let count = match &count[0] {
-                    IonValue::Struct(data) => match data.get("_1") {
-                        Some(IonValue::Integer(count)) => count,
-                        _ => panic!("First count returned a non integer"),
-                    },
+                let count = match &count[0].get("_1") {
+                    Some(IonValue::Integer(count)) => count,
```
Comment (aegis22) [inline tests/transaction.rs:76]: This code is the same as in lines 112, 113, maybe it could be a function to get a count.

**Review by aegis22 — COMMENTED** [review]
Some things for you, @DavidBM  :)

**Comment by aegis22** [comment]
Delete printlns

File: src/cursor.rs (hunk 1 above)
Comment (DavidBM) [reply src/cursor.rs:106]: Nope. You can get a page (which would fulfill the condition "Some(values)") which doesn't contains any next_page_token. 

Actually I added that extra condition when doing the testing. When you call "load_more" it updates the next_page property. You must check that in order to see if there are more pages available. If not, you are doing an extra call to QLDB for nothing (returning 0 results or an error if you send an empty token).

File: tests/cursor.rs (hunk 2 above)
Comment (DavidBM) [reply tests/cursor.rs]: Yep! Updating!

File: tests/transaction.rs (hunk 4 above)
Comment (DavidBM) [reply tests/transaction.rs:76]: Yep, and it isn't needed because now we have the method `.count()`. I will replace by it.

**Comment by DavidBM** [comment]
The only println that I see are inside comment blocks

**Comment by DavidBM** [comment]
@aegis22 Code review changes applied.

File: src/cursor.rs (hunk 1 above)
Comment (aegis22) [reply src/cursor.rs:106]: Okk.

**Review by aegis22 — APPROVED** [review]

**Pull request merged**